apps/server/src/services/TimerService.ts

```diff
@@ -302,6 +311,37 @@ export class TimerService {
           this.pausedTime,
           this.timer.clock,
         );
+
+        if (
+          this.playback === 'play' &&
+          this.timer.endAction === EndAction.Stop &&
+          this.timer.current <= 0 &&
+          this.timer.finishedAt === null
+        ) {
+          this.timer.finishedAt = this.timer.clock;
+          PlaybackService.stop();
+        }
+
+        if (
+          this.playback === 'play' &&
+          this.timer.endAction === EndAction.Pause &&
+          this.timer.current <= 0 &&
+          this.timer.finishedAt === null
+        ) {
+          this.timer.finishedAt = this.timer.clock;
+          PlaybackService.pause();
+        }
+
+        if (
+          this.playback === 'play' &&
+          this.timer.endAction === EndAction.Next &&
+          this.timer.current <= 0 &&
+          this.timer.finishedAt === null
+        ) {
+          this.timer.finishedAt = this.timer.clock;
+          PlaybackService.startNext();
+        }
+
```

Comment:
I still think that all of this should be handled as a side effect in `_onFinish()`

Reply:
by moving the logic to the `_onFinish()` function it causes the timer to go negative before my logic get's handled
e.g by choosing pause it pauses at -00:00:01

Reply:
Hmm. That sounds plausible.
The timer updates once a second and assumes a finished timer has gone past the 0

So this means we might be reading the timer at -300ms which gets rounded to -1s

The only other way is to have a 0 seconds and a -0 seconds which would be worse I guess

I short: the logic is doing what is supposed to do, although the presentation looks strange

We still need to go through the function to handle other side effects like automation sync.

What do you think?
If we think that the -1 hurts the confidence of the user to the timer,  my instinct would be to cut the pause functionality, potentially replace with a play next - load next

Reply:
I think for the end user it looks strange if the timer doesn't stop at 0. I agree with your suggestion and will remove the pause functionality and also expand the functionality to load-next and play-next